### ISSUE TITLE
`yarn encore` compiles successfully and then, much much later, throws an error

### ISSUE THREAD

**Issue by danbettles** [symfony/webpack-encore#1146]
Good morning

I'm a bit stumped and would like some help, if possible, please.  Any pointers will be very welcome, thank you.  I haven't been able to find anything of use elsewhere.

`yarn encore` apparently hangs after saying it's successfully compiled our project.  After a couple attempts, I just left it running and eventually it threw an error.  I don't know enough to understand what the stack trace is pointing at.  It looks as if the problem _could_ be webpack but I can't find reference to the error code.  The error code doesn't appear to have anything to do with Yarn.  I wonder, have _you_ seen this problem before?  Is there any way to see what Encore is doing?  I've read about, and tried, Encore's `--profile` CLI option but nothing new is displayed and I don't see a log file.  I'm not sure where to begin with this.

Here's what I saw the last time I tried running Encore:

```sh
seetheworld@dan:~/vhtdocs/s/stw-bookings$ yarn --verbose encore dev --profile
yarn run v1.22.17
verbose 0.648418183 Checking for configuration file "/home/seetheworld/vhtdocs/s/stw-bookings/.npmrc".
verbose 0.649559572 Checking for configuration file "/home/seetheworld/.npmrc".
verbose 0.650454348 Checking for configuration file "/home/seetheworld/.nvm/versions/node/v14.18.0/etc/npmrc".
verbose 0.651247908 Checking for configuration file "/home/seetheworld/vhtdocs/s/stw-bookings/.npmrc".
verbose 0.6520772 Checking for configuration file "/home/seetheworld/vhtdocs/s/.npmrc".
verbose 0.653013165 Checking for configuration file "/home/seetheworld/vhtdocs/.npmrc".
verbose 0.654031989 Checking for configuration file "/home/seetheworld/.npmrc".
verbose 0.654781117 Checking for configuration file "/home/.npmrc".
verbose 0.65640441 Checking for configuration file "/home/seetheworld/vhtdocs/s/stw-bookings/.yarnrc".
verbose 0.657429162 Checking for configuration file "/home/seetheworld/.yarnrc".
verbose 0.658158477 Found configuration file "/home/seetheworld/.yarnrc".
verbose 0.659509905 Checking for configuration file "/home/seetheworld/.nvm/versions/node/v14.18.0/etc/yarnrc".
verbose 0.660365284 Checking for configuration file "/home/seetheworld/vhtdocs/s/stw-bookings/.yarnrc".
verbose 0.661126718 Checking for configuration file "/home/seetheworld/vhtdocs/s/.yarnrc".
verbose 0.662244781 Checking for configuration file "/home/seetheworld/vhtdocs/.yarnrc".
verbose 0.663180621 Checking for configuration file "/home/seetheworld/.yarnrc".
verbose 0.664047646 Found configuration file "/home/seetheworld/.yarnrc".
verbose 0.679342779 Checking for configuration file "/home/.yarnrc".
verbose 0.693341934 current time: 2022-08-20T08:43:27.996Z
$ /home/seetheworld/vhtdocs/s/stw-bookings/node_modules/.bin/encore dev --profile
Running webpack ...

 DONE  Compiled successfully in 60850ms                                                                                                                                                             08:44:34

135 files written to web/build/js

<--- Last few GCs --->

[21710:0x4a52810]   688427 ms: Mark-sweep (reduce) 1939.5 (1971.4) -> 1938.8 (1971.4) MB, 6465.8 / 0.0 ms  (average mu = 0.079, current mu = 0.002) allocation failure scavenge might not succeed
[21710:0x4a52810]   694602 ms: Mark-sweep (reduce) 1939.8 (1968.4) -> 1938.9 (1969.7) MB, 6133.8 / 0.0 ms  (average mu = 0.044, current mu = 0.007) allocation failure scavenge might not succeed


<--- JS stacktrace --->

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0xa389b0 node::Abort() [webpack]
 2: 0x96e0af node::FatalError(char const*, char const*) [webpack]
 3: 0xbb7a4e v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [webpack]
 4: 0xbb7dc7 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [webpack]
 5: 0xd73fd5  [webpack]
 6: 0xd74b5f  [webpack]
 7: 0xd8299b v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [webpack]
 8: 0xd8655c v8::internal::Heap::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [webpack]
 9: 0xd54c3b v8::internal::Factory::NewFillerObject(int, bool, v8::internal::AllocationType, v8::internal::AllocationOrigin) [webpack]
10: 0x109d21f v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned long*, v8::internal::Isolate*) [webpack]
11: 0x1446379  [webpack]
Aborted
verbose 701.761060613 Error: Command failed with exit code 134.
    at ProcessTermError.ExtendableBuiltin (/home/seetheworld/.nvm/versions/node/v14.18.0/lib/node_modules/yarn/lib/cli.js:721:66)
    at ProcessTermError.MessageError (/home/seetheworld/.nvm/versions/node/v14.18.0/lib/node_modules/yarn/lib/cli.js:750:123)
    at new ProcessTermError (/home/seetheworld/.nvm/versions/node/v14.18.0/lib/node_modules/yarn/lib/cli.js:790:113)
    at /home/seetheworld/.nvm/versions/node/v14.18.0/lib/node_modules/yarn/lib/cli.js:34672:30
    at Generator.throw (<anonymous>)
    at step (/home/seetheworld/.nvm/versions/node/v14.18.0/lib/node_modules/yarn/lib/cli.js:310:30)
    at /home/seetheworld/.nvm/versions/node/v14.18.0/lib/node_modules/yarn/lib/cli.js:323:13
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
error Command failed with exit code 134.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Here are some environmental details:

In `composer.json`:
symfony/webpack-encore-bundle: ^1.3

In `package.json`:
@symfony/webpack-encore: ^3.0.0

node v14.18.0
yarn 1.22.17

**Comment by Lyrkan** [comment]
Hi @danbettles,

Would you happen to have build notifications enabled (`Encore.enableBuildNotifications()` in your `webpack.config.js` file) by any chance?

If so, try disabling them and check if the issue still happens.

**Comment by danbettles** [comment]
Many thanks for your response.  I'll give that a go.

**Comment by danbettles** [comment]
@Lyrkan (Unfortunately) we're not using that feature.  I've posted an issue targeting webpack more, to see if anyone on that side has any ideas.  Thanks again.

**Comment by Kocal** [comment]
@danbettles how did you solve it?

**Comment by danbettles** [comment]
Hello, @Kocal 

I haven't solved the problem yet, but I'm getting closer to the source.  I stripped everything back to the bare minimum and then built it up again.  The problem appears to be in/around PostCSS: we use the autoprefixer plugin.  I need to experiment in that area now.

It appeared the problem wasn't with Encore, so I posted [a question on stackoverflow.com](https://stackoverflow.com/questions/73441626/webpack-via-encore-compiles-code-successfully-and-then-many-many-minutes-lat).  I'll update that if/when I figure it out 😉 

Thanks for your interest.
